res/controllers/M-Audio-Xponent-Advanced-scripts.js

```diff
@@ -0,0 +1,845 @@
+    midi.sendShortMsg(deck.on, MaudioXponent.leds.scratch, deck.scratchEnabled);
+};
+
+MaudioXponent.playlist = function(channel, control, value, status, group) {
+    var deck = MaudioXponent.getDeck(group);
+    switch (control) {
+    case 28:
+        engine.setValue("[Playlist]", "SelectPrevTrack", 1);
+        midi.sendShortMsg(deck.on, control, true);
+	break;
+    case 29:
+        engine.setValue("[Playlist]", "SelectNextTrack", 1);
+        midi.sendShortMsg(deck.on, control, true);
+	break;
+    case 30:
+        var activenow = engine.getValue(deck.group, "play");
+        if (activenow == 1) {    // If currently active
+            engine.setValue("[Playlist]", "LoadSelectedIntoFirstStopped", 1);
+        } else {
+            engine.setValue(deck.group, "LoadSelectedTrack", 1);
+        }
+        midi.sendShortMsg(deck.on, control, true);
+	break;
+    case 31:
+        engine.setValue("[Playlist]", "SelectPrevPlaylist", 1);
+        midi.sendShortMsg(deck.on, control, true);
+	break;
+    case 32:
+        engine.setValue("[Playlist]", "SelectNextPlaylist", 1);
+        midi.sendShortMsg(deck.on, control, true);
+	break;
+    }
+};
+
+MaudioXponent.playlistoff = function(channel, control, value, status, group) {
+    var deck = MaudioXponent.getDeck(group);
+    midi.sendShortMsg(deck.on, control, false);
+};
+
+MaudioXponent.keyLock = function(channel, control, value, status, group) {
+    //script.midiDebug(channel, control, value, status);
+    var deck = MaudioXponent.getDeck(group);
+ 
+    if (!deck.shift) {
+        // Unshifted = Keylock
+        var currentValue = engine.getParameter(deck.group, "keylock");
+        if (currentValue == 0){
+            engine.setValue(deck.group,"keylock",0x01);
+        } else {
+            engine.setValue(deck.group,"keylock", 0x00);
+        }
+    }else{        
+        // Shifted = Quantize
+        var currentValue = engine.getParameter(deck.group, "quantize");
+        if (currentValue == 0){
+            engine.setValue(deck.group,"quantize",0x01);
+        } else {
+            engine.setValue(deck.group,"quantize", 0x00);
+        }
+    }    
+};
+
+MaudioXponent.onkeyLock = function(value, group) {
+    var deck = MaudioXponent.getDeck(group);
+    midi.sendShortMsg(deck.on, MaudioXponent.leds.key, value);
+};
+
+MaudioXponent.brake = function(channel, control, value, status, group) {
+    //script.midiDebug(channel, control, value, status, group);
+    var deck = MaudioXponent.getDeck(group);
+    var activate = (status == deck.on);
+    engine.brake(deck.id, activate);
+};
+
+MaudioXponent.cue = function(channel, control, value, status, group) {
+    //script.midiDebug(channel, control, value, status);
+    var deck = MaudioXponent.getDeck(group);
+    var activate = (status == deck.on);
+    engine.setValue(deck.group, "cue_default", activate);
+    
+    //TODO: Is this needed?
+    midi.sendShortMsg(deck.on, control, activate);
```

Comment:
I could map this in script, but the only thing I would gain is the ability to turn the led on and off if you held down the button while switching banks (1&2 to 3&4), but you'd still have the light turn on when you switch back, so it could still get out of sync possibly, and the cue button is only ever momentary anyway. It didn't seem worth the trouble. At this point, Cue, and seek fwd and back are the only things still handled by the XML, which seems adequate to me.